dist/tools/flake8/check.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2017 Alexandre Abadie <alexandre.abadie@inria.fr>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if tput colors &> /dev/null && [ $(tput colors) -ge 8 ]; then
+    CERROR="\e[1;31m"
+    CWARNING="\e[1;33m"
+    CRESET="\e[0m"
+else
+    CERROR=
+    CWARNING=
+    CRESET=
+fi
+
+DIST_TOOLS=${RIOTBASE:-.}/dist/tools
+
+. ${DIST_TOOLS}/ci/changed_files.sh
+
+FILES=$(FILEREGEX='*.py$' changed_files)
+
+if [ -z "${FILES}" ]
+then
+    exit 0
+fi
+
+ERRORS=$(flake8 --config=${DIST_TOOLS}/flake8/flake8.cfg ${FILES})
+
+if [ -n "${ERRORS}" ]
+then
+    printf "${CWARNING}There are style issues in the following Python scripts:${CRESET}\n\n"
+    printf "${ERRORS}\n"
+    exit 0
```

Comment:
just to make this more obvious: this check script never fails but just list problematic files with a warning yellow color. I can change to a more strict rule if desired: red error color + exit 1.

Reply:
mhm, did you run it all over RIOT checking all python in the current master? If that's the case and all current python stuff sticks with rules, I'd says lets be more strict and `exit 1` because most people don't look into build logs if they succeed.

Reply:
> did you run it all over RIOT checking all python in the current master?

Yes and the scripts in test should be fine but not the one in the dist/tools directory. But I agree we should be more strict now and open PRs if problems occur later.